tools/run_tests/xds_k8s_test_driver/framework/infrastructure/k8s.py

```diff
@@ -78,9 +79,92 @@ class PortForwardingError(Exception):
     """Error forwarding port"""
 
 
+class PortForwarder:
+    PORT_FORWARD_LOCAL_ADDRESS: str = '127.0.0.1'
+
+    def __init__(self,
+                 context: str,
+                 namespace: str,
+                 destination: str,
+                 remote_port: int,
+                 local_port: Optional[int] = None,
+                 local_address: Optional[str] = None):
+        self.context = context
+        self.namespace = namespace
+        self.destination = destination
+        self.remote_port = remote_port
+        self.local_address = local_address or self.PORT_FORWARD_LOCAL_ADDRESS
+        self.local_port, self.subprocess = self._establish_port_forwarding(
```

Comment:
Normally I'd not be doing any heavy work in the constructor, and have separate `connect` (so class can be instantiated for unit tests and such), but it doesn't matter as much in this case.

Reply:
Done.